File: src/main/java/one/nxeu/vlcore/core/helper/ServerInfoHelper.java
package one.nxeu.vlcore.core.helper;

import net.minecraft.server.MinecraftServer;

public class ServerInfoHelper {

    public static long[] getRecentMspt(MinecraftServer server) {
        return server.k;
    }

    public static double[] getRecentTps(MinecraftServer server) {
        return server.recentTps;
    }

    public static double getTps(MinecraftServer server) {
        return TpsHelper.average(getRecentTps(server));
    }

    public static double getMspt(MinecraftServer server) {
        return TpsHelper.toMilliseconds(TpsHelper.average(ServerInfoHelper.getRecentMspt(server)));
    }
}

File: src/main/java/one/nxeu/survivalcore/helper/ServerInfoHelper.kt
package one.nxeu.survivalcore.helper

import net.minecraft.server.MinecraftServer
import org.bukkit.Server
import org.bukkit.craftbukkit.v1_19_R3.CraftServer

class ServerInfoHelper {
    companion object {
        fun getRecentMspt(server: Server): LongArray {
            val minecraftServer: MinecraftServer = (server as CraftServer).server
            return minecraftServer.tickTimes
        }
        fun getRecentTps(server: Server): DoubleArray {
            val minecraftServer: MinecraftServer = (server as CraftServer).server
            return minecraftServer.recentTps
        }

        fun getMspt(server: Server): Double {
            return TickHelper.toMilliseconds(getRecentMspt(server).average())
        }
        fun getTps(server: Server): Double {
            return TickHelper.toMilliseconds(getRecentTps(server).average())
        }
    }
}